### PR TITLE
fix: s3 endpoint url field while importing json

### LINKED ISSUE #29408: [Bug]: S3 datasource screen in import modal has an additional End Point URL field
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

When I import a json which contains S3 DS, an additional End point url field is added. 

### Steps To Reproduce

RELEASE:
release

PROD:
prod

### Public Sample App

_No response_

### Environment

Release

### Issue video log

_No response_

### Version

Cloud

### PR DESCRIPTION
**Pull request by NilanshBansal**
## Description

Before the merge of [this](https://github.com/appsmithorg/appsmith/pull/28146) PR, the entire `datasourceConfiguration` object in the exported app json was set as null. 
Example JSON for s3 plugin earlier : 
```
{
  "datasourceList": [
    {
      "name": "S3",
      "pluginId": "amazons3-plugin",
      "messages": [],
      "isAutoGenerated": false,
      "deleted": false,
      "gitSyncId": ""
    }
  ]
}
```

Post the merge, the `datasourceConfiguration.properties` field was removed and the `datasourceConfiguration.endpoints` field was kept populated.
JSON for s3 plugin post the change :

```
{
  "datasourceList": [
    {
      "datasourceConfiguration": {
        "endpoints": [
          {
            "host": ""
          }
        ]
      },
      "name": "Untitled datasource 1",
      "pluginId": "amazons3-plugin",
      "messages": [],
      "isAutoGenerated": false,
      "deleted": false,
      "gitSyncId": ""
    }
  ]
}
```

Due to the above change the S3 plugin form.json condition [here](https://github.com/appsmithorg/appsmith/blob/012a69296cdf59cd7874941f94701d24da498b44/app/server/appsmith-plugins/amazons3Plugin/src/main/resources/form.json#L75) started displaying the field. 

**This was also due to a combination of a pre-existing error in the form.json, which on importing the app json showed the first dropdown value from the S3 service provider, but did not initialised the value. 
On changing the dropdown the value was automatically assigned.**

https://github.com/appsmithorg/appsmith/assets/25542733/78d5f693-11fc-4bd6-8cd7-90ee300b2b58

> This PR updates the form JSON to initialise the default value which leads to the amazon-s3 plugin getting auto-initialised with some default values.

#### PR fixes following issue(s)
Fixes #29408 

#### Type of change
- Bug fix (non-breaking change which fixes an issue)

- [x] Manual
- [ ] JUnit
- [ ] Jest
- [ ] Cypress
>
>
#### Test Plan
1. Imported a recent json with S3 DS [created on 1 Dec] and tested for the issue
2. Imported an older than a month json with S3 DS and tested for the issue.
3. Exported the current imported file, and re-imported that into a new workspace and tested for the issue
4. Forked the json into a new workspace and tested
5. Did an app level import of a latest json and tested for the issue
6. Forked from inside the application and tested for the issue
>
>
#### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

## Summary by CodeRabbit

- **Refactor**
  - Improved the handling of sensitive information in data source configurations to enhance security.

- **Tests**
  - Updated tests to align with the new logic for data source configuration properties.

### CODE REVIEW

File: app/server/appsmith-server/src/main/java/com/appsmith/server/datasources/exports/DatasourceExportableServiceCEImpl.java [hunk 1]
```diff
@@ -107,7 +107,6 @@ private void removeSensitiveFields(DatasourceStorage datasourceStorage) {
             datasourceStorage.getDatasourceConfiguration().setAuthentication(null);
             datasourceStorage.getDatasourceConfiguration().setSshProxy(null);
             datasourceStorage.getDatasourceConfiguration().setSshProxyEnabled(null);
-            datasourceStorage.getDatasourceConfiguration().setProperties(null);
```
Comment (sneha122) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/datasources/exports/DatasourceExportableServiceCEImpl.java:110]: @NilanshBansal I hope we have checked for all plugins properties to ensure that we are not exposing any sensitive information by not making it null

Comment (NilanshBansal) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/datasources/exports/DatasourceExportableServiceCEImpl.java:110]: I am keeping this set as null @sneha122

**Pull request merged**